### PR TITLE
feat(collections): Get all the collections related to a Entity

### PR DESCRIPTION
**Pull request by akashgp09**
### Problem

This PR Fixes : [BB-614](https://tickets.metabrainz.org/projects/BB/issues/BB-614?filter=allissues)
Follow up from PR : [#599](https://github.com/bookbrainz/bookbrainz-site/pull/599)
### Solution

Added `collections()` to all the **Entity Model**. This helps to fetch all the collections **related** to the entity.
Also added the corresponding **tests** for it.

### Areas of Impact

- [models/entities/author.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/src/models/entities/author.js)
- [models/entities/work.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/src/models/entities/work.js)
- [models/entities/edition.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/src/models/entities/edition.js)
- [models/entities/editionGroup.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/src/models/entities/editionGroup.js)
- [models/entities/publisher.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/src/models/entities/publisher.js)
- [test/testAuthor.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/test/testAuthor.js)
- [test/testWork.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/test/testWork.js)
- [test/testEdition.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/test/testEdition.js)
- [test/testEditionGroupjs](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/test/testEditionGroup.js)
- [test/testPublisher.js](https://github.com/bookbrainz/bookbrainz-data-js/blob/master/test/testPublisher.js)

### CODE REVIEW

**Review by MonkeyDo — COMMENTED** [review]
Great work, that looks clean and I tested locally with the other PR, everything is working fine!
The tests look good too; I have minor nitpicks below:

File: test/testAuthor.js [hunk 1]
```diff
@@ -211,4 +214,67 @@ describe('Author model', () => {
...
+		await new UserCollection(userCollectionAttribs).save(null, {method: 'insert'});
+		await new UserCollectionItem(userCollectionItemAttribs).save(null, {method: 'insert'});
+		await model.refresh({
+			withRelated: ['collections']
+		});
+		const json = model.toJSON();
+		const {collections} = json;
+
+		// collections exist
+		return expect(collections).to.have.lengthOf.above(0);
```
[the remark this answers is not in the document]
Comment (akashgp09) [reply test/testAuthor.js:253]: >Why not compare the expected length directly? I think that would make it a more precise test:

 Yes Indeed, will make the changes :)

**Review by MonkeyDo — APPROVED** [review]
Thanks ! I'll merge and publish a new minor version of the package that you can then add to https://github.com/bookbrainz/bookbrainz-site/pull/599.

You'll need to commit both the package.json and package-lock.json files.

**Comment by MonkeyDo** [comment]
👍 published `bookbrainz-data@2.10.0`

**Pull request merged**